src/detect-engine-address.c

```diff
@@ -1009,6 +1010,38 @@ static int DetectAddressParse2(const DetectEngineCtx *de_ctx,
     return -1;
 }
 
+/**
+ * \internal
+ * \brief Wrapper function for address parsing to minimize heap allocs during address parsing.
+ *
+ * \retval Return value from DetectAddressParse2
+ */
+static int DetectAddressParse2(const DetectEngineCtx *de_ctx, DetectAddressHead *gh,
+        DetectAddressHead *ghn, const char *s, int negate, ResolvedVariablesList *var_list,
+        int recur)
+{
+    int rc;
+#define MAX_ADDRESS_LENGTH 8192
+
+    size_t address_length = strlen(s);
+    if (address_length > (MAX_ADDRESS_LENGTH - 1)) {
+        char *address = SCCalloc(1, address_length);
+        if (address == NULL) {
+            SCLogError(SC_ERR_ADDRESS_ENGINE_GENERIC, "Unable to allocate"
+                                                      " memory for address parsing.");
+            return -1;
+        }
+        rc = DetectAddressParseInternal(
+                de_ctx, gh, ghn, s, negate, var_list, recur, address, address_length);
+        SCFree(address);
+    } else {
+        char address[MAX_ADDRESS_LENGTH] = "";
```

Comment:
Is this alloc saving with the code complexity ?

Reply:
The allocation is used with strings exceeding the threshold; the common case is expected to be strings with length under the threshold and thus, a stack allocation.

Reply:
I understand this.

The question is about the trade-off between code complexity and the performance (saving one allocation...)

Why do we need this `address` in the first place ? (we have `s`)